torch/utils/data/datapipes/utils/common.py

```diff
@@ -345,9 +347,9 @@ def autoclose(self):
         Close steam if there is no children, or make it to be automatically closed as soon as
         all child streams are closed.
         """
+        self.close_on_last_child = True
         if self.child_counter == 0:
             self.close()
-        self.close_on_last_child = True
```

Comment:
hmm.. why originally set `close_on_last_child` after closing it...

Reply:
to clarify, i mean the new implementation looks correct, just don't understand the original implementation~